Svelte 5: Parsing Error with Snippets

### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.56.0

### What version of `eslint-plugin-svelte` are you using?

2.63.0-next.5

### What did you do?

ESlint fails to parse snippets whose arguments are function calls.
```html
<script lang="ts">
	import type { Snippet } from "svelte";

	let { foo } = $props<{ foo: Snippet<[string]> }>();

	function bar() {
		return "baz";
	}
</script>

{@render foo(bar())}
```

### What did you expect to happen?

Parsing should not fail.

https://github.com/sveltejs/eslint-plugin-svelte/assets/88583085/832869b9-2b9d-4a1f-b598-2972416e82c3

### What actually happened?

```bash
  13:11  error  Parsing error: ')' expected

✖ 1 problem (1 error, 0 warnings)
```


### Link to **GitHub Repo** with Minimal Reproducible Example

https://github.com/abdel-17/svelte-eslint-bug-reproduction

### Additional comments

_No response_

--- a/.changeset/twenty-trainers-peel.md
+++ b/.changeset/twenty-trainers-peel.md
@@ -1,0 +1,5 @@
+---
+"eslint-plugin-svelte": patch
+---
+
+feat: update parser

--- a/.eslintignore
+++ b/.eslintignore
@@ -35,3 +35,4 @@
 
 # Parser bug?
 /tests/fixtures/rules/indent/invalid/const-tag01-input.svelte
+/tests/fixtures/rules/indent/invalid/const-tag01-output.svelte

--- a/.eslintignore
+++ b/.eslintignore
@@ -32,3 +32,6 @@
 
 # JSONSchema bug?
 /.devcontainer/devcontainer.json
+
+# Parser bug?
+/tests/fixtures/rules/indent/invalid/const-tag01-input.svelte

--- a/package.json
+++ b/package.json
@@ -78,7 +78,7 @@
     "postcss-safe-parser": "^6.0.0",
     "postcss-selector-parser": "^6.0.11",
     "semver": "^7.5.3",
-    "svelte-eslint-parser": ">=0.34.0-next.4 <1.0.0"
+    "svelte-eslint-parser": ">=0.34.0-next.9 <1.0.0"
   },
   "devDependencies": {
     "@1stg/browserslist-config": "^2.0.0",

--- a/package.json
+++ b/package.json
@@ -59,7 +59,7 @@
   },
   "peerDependencies": {
     "eslint": "^7.0.0 || ^8.0.0-0 || ^9.0.0-0",
-    "svelte": "^3.37.0 || ^4.0.0 || ^5.0.0-next.33"
+    "svelte": "^3.37.0 || ^4.0.0 || ^5.0.0-next.68"
   },
   "peerDependenciesMeta": {
     "svelte": {
@@ -166,7 +166,7 @@
     "stylelint": "^16.0.0",
     "stylelint-config-standard": "^36.0.0",
     "stylus": "^0.62.0",
-    "svelte": "^5.0.0-next.33",
+    "svelte": "^5.0.0-next.68",
     "svelte-adapter-ghpages": "0.2.2",
     "svelte-i18n": "^4.0.0",
     "tslib": "^2.5.0",

--- a/src/rules/indent-helpers/svelte.ts
+++ b/src/rules/indent-helpers/svelte.ts
@@ -227,7 +227,7 @@ export function defineVisitor(context: IndentContext): NodeListener {
 			const rightParenToken = sourceCode.getTokenBefore(sourceCode.getLastToken(node));
 
 			offsets.setOffsetToken(leftParenToken, 1, calleeToken);
-			offsets.setOffsetElementList([node.argument], leftParenToken, rightParenToken, 1);
+			offsets.setOffsetElementList(node.arguments, leftParenToken, rightParenToken, 1);
 		},
 		// ----------------------------------------------------------------------
 		// BLOCKS
@@ -488,19 +488,22 @@ export function defineVisitor(context: IndentContext): NodeListener {
 			offsets.setOffsetToken(id.firstToken, 1, snippetToken);
 
 			const leftParenToken = sourceCode.getTokenBefore(
-				node.context || sourceCode.getLastToken(node),
+				node.params[0] || sourceCode.getLastToken(node),
 				{
 					filter: isOpeningParenToken,
 					includeComments: false
 				}
 			)!;
 
-			const rightParenToken = sourceCode.getTokenAfter(node.context || leftParenToken, {
-				filter: isClosingParenToken,
-				includeComments: false
-			})!;
+			const rightParenToken = sourceCode.getTokenAfter(
+				node.params[node.params.length - 1] || leftParenToken,
+				{
+					filter: isClosingParenToken,
+					includeComments: false
+				}
+			)!;
 			offsets.setOffsetToken(leftParenToken, 1, id.firstToken);
-			offsets.setOffsetElementList([node.context], leftParenToken, rightParenToken, 1);
+			offsets.setOffsetElementList(node.params, leftParenToken, rightParenToken, 1);
 		},
 		// ----------------------------------------------------------------------
 		// COMMENTS

--- a/src/rules/mustache-spacing.ts
+++ b/src/rules/mustache-spacing.ts
@@ -400,10 +400,13 @@ export default createRule('mustache-spacing', {
 			},
 			SvelteSnippetBlock(node) {
 				const openBlockOpeningToken = sourceCode.getFirstToken(node);
-				const openBlockClosingToken = sourceCode.getTokenAfter(node.context || node.id, {
-					includeComments: false,
-					filter: isClosingBraceToken
-				})!;
+				const openBlockClosingToken = sourceCode.getTokenAfter(
+					node.params[node.params.length - 1] || node.id,
+					{
+						includeComments: false,
+						filter: isClosingBraceToken
+					}
+				)!;
 				verifyBraces(
 					openBlockOpeningToken,
 					openBlockClosingToken,

--- a/tests/fixtures/rules/indent/invalid/const-tag01-requirements.json
+++ b/tests/fixtures/rules/indent/invalid/const-tag01-requirements.json
@@ -1,0 +1,4 @@
+{
+	"FIXME": "Bug in Svelte v5-next.68?",
+	"svelte": "^4 || ^3"
+}

--- a/tests/utils/utils.ts
+++ b/tests/utils/utils.ts
@@ -132,15 +132,23 @@ export function loadTestCases(
 			}
 			const errors = fs.readFileSync(errorFile, 'utf8');
 			config.errors = parseYaml(errors);
+			for (const error of config.errors) {
+				if (error.suggestions) {
+					error.suggestions = error.suggestions?.map((s: any) => ({
+						output: s.output,
+						desc: s.desc
+					}));
+				}
+			}
 			if (fixable) {
-				let output = null;
+				let output;
 				try {
 					output = fs.readFileSync(outputFile, 'utf8');
 				} catch (_e) {
 					writeFixtures(ruleName, inputFile);
 					output = fs.readFileSync(outputFile, 'utf8');
 				}
-				config.output = output;
+				config.output = output === config.code ? null : output;
 			}
 
 			return config;

--- a/tests/utils/utils.ts
+++ b/tests/utils/utils.ts
@@ -133,7 +133,7 @@ export function loadTestCases(
 			const errors = fs.readFileSync(errorFile, 'utf8');
 			config.errors = parseYaml(errors);
 			if (fixable) {
-				let output;
+				let output = null;
 				try {
 					output = fs.readFileSync(outputFile, 'utf8');
 				} catch (_e) {

